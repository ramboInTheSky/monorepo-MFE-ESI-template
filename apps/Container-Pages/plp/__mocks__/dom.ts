export function createElementWithChildren(totalChildren: number, amendChild?: (child: HTMLElement, i: number) => any) {
    const container = document.createElement("div")
    for (let i = 0; i < totalChildren; i += 1) {
        const child = document.createElement("div")
        jest.spyOn(child, "getBoundingClientRect")
        child.appendChild(document.createTextNode(`child ${i + 1}`))
        if (amendChild) amendChild(child, i)
        container.appendChild(child)
    }
    return container
}

export function mockWindowLocation(props = {}) {
    const {location} = window

    delete (window as any).location
    ;(window as any).location = {...location, ...props}

    const restore = () => {
        ;(window as any).location = location
    }

    return restore
}

export function mockWindowScrollTo() {
    const {scrollTo} = window
    delete (window as any).scrollTo
    window.scrollTo = jest.fn()
    return () => {
        window.scrollTo = scrollTo
    }
}
