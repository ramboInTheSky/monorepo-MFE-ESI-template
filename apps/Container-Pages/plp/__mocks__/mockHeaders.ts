import {TestDataSwitcher} from "../cypress/modules/filter"

export const amidoHeaders = TestDataSwitcher(
    ["postdeploy"],
    {
        "x-monorepo-language": "en",
        "x-monorepo-realm": "amido",
        "x-monorepo-territory": "gb",
        "x-set-caching": "false",
    },
    {
        "x-monorepo-language": "en",
        "x-monorepo-realm": "amido",
        "x-monorepo-territory": "gb",
        "test-with-local-esi": true,
    },
)
