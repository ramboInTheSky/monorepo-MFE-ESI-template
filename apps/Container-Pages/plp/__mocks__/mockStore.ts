import thunk from "redux-thunk"
import configureMockStore from "redux-mock-store"
import {ThemeColor, mockColors} from "../../../../libs/framework/themes/src"
import State from "../src/models/State"
import themeDimensions from "./themeDimensions.json"
import TextAlignment from "../src/models/textAlignment"
import {Dimensions} from "../src/models/Dimensions"
import {TextModel} from "../src/models/Text"
import {Theme} from "../src/models/theme"
import {SearchApiRequestTypes} from "../src/config/constants"
import {RequestDuckState} from "../src/ducks/request/types"
import text from "./default-text.json"

export const mockColours: ThemeColor = mockColors
export const mockDimensions: Dimensions = themeDimensions
export const mockText: TextModel = text

export const mockState: State = {
    text,
    textAlignment: TextAlignment.Ltr,
    request: {
        headers: {"x-monorepo-territory": "gb", "x-monorepo-language": "en"},
        url: "/",
        siteUrl: "",
        searchTerm: null,
        category: null,
        gender: null,
        territoryName: "",
        page: 6,
        type: SearchApiRequestTypes.Category,
        useDevEsi: false,
        viewportSize: "desktop",
        bloomreachPersonalizationEnabled: false,
    },
    search: {
        seoFilters: "",
        singleOptionFacetList: ["personalised", "feat", "sizetype", "deliverby"],
        searchBannerHtml: null,
        totalResults: 1234,
        filters: {
            Test1: {
                name: "Test1",
                facets: ["opt1"],
                displayName: "Test 1",
                type: "feat",
                isFilterOpen: true,
                isViewMoreOpen: true,
            },
            Test2: {
                name: "Test2 with options",
                facets: ["opt2"],
                displayName: "Test 2",
                type: "filter",
                isFilterOpen: false,
                isViewMoreOpen: false,
            },
            Test3: {
                name: "Test2 with options",
                displayName: "",
                type: "price",
                isFilterOpen: false,
                max: 100,
                min: 0,
                selectedMax: 70,
                selectedMin: 30,
                currencyCode: "GBP",
            },
            brand: {
                name: "Brand",
                facets: [
                    "brand:znth",
                    "brand:one",
                    "brand:oops",
                    "brand:two",
                    "brand:tcas",
                    "brand:three",
                    "brand:four",
                    "brand:five",
                    "brand:six",
                    "brand:2 size",
                ],
                displayName: "Test 1",
                type: "filter",
                isFilterOpen: true,
                isViewMoreOpen: true,
            },
            Price: {
                name: "Test2 with options",
                displayName: "",
                type: "price",
                isFilterOpen: false,
                max: 100,
                min: 0,
                selectedMax: 70,
                selectedMin: 30,
                currencyCode: "GBP",
            },
        },
        filtersSort: ["Test1", "Test2"],
        facets: {
            opt1: {n: "opt1", c: 1, v: "opt1", incompatibleWith: [], d: false},
            opt2: {n: "opt2", c: 1, v: "opt2", incompatibleWith: ["opt3"], d: false},
            opt3: {n: "opt3", c: 1, v: "opt3", incompatibleWith: ["opt2"], d: false},
        },
        items: [
            {itemNumber: "111111", newIn: true},
            {itemNumber: "222222", newIn: true},
        ],
        sorting: {
            selected: "score",
            options: [
                {
                    name: "Most Relevant",
                    value: "score",
                },
                {
                    name: "Most Popular",
                    value: "newpop",
                },
                {
                    name: "Alphabetical",
                    value: "alpha",
                },
                {
                    name: "Price: Low - High",
                    value: "pricelow",
                },
                {
                    name: "Price: High - Low",
                    value: "pricehi",
                },
            ],
        },
        startPage: 1,
        endPage: 1,
        isFetchingNextPage: false,
        isFetchingPreviousPage: false,
        isFetchingPageItems: false,
        debounceTime: 300,
        shouldFacetsBeFixed: false,
        autoCorrectQuery: null,
        title: "",
        initialItemsPerPage: 8,
        fetchTriggerOffset: {
            xs: 4,
            sm: 4,
            md: 6,
            lg: 6,
            xl: 8,
        },
        itemsPerRow: {
            inPageFiltersEnabled: {
                xs: 4,
                sm: 4,
                md: 6,
                lg: 6,
                xl: 8,
            },
            default: {
                xs: 4,
                sm: 4,
                md: 6,
                lg: 6,
                xl: 8,
            },
        },
        currentBreakpoint: "md",
        fetchTriggerOffsetRows: {
            xs: 2,
            sm: 2,
            md: 2,
            lg: 2,
            xl: 2,
        },
        subsequentPagesNonLazyloadRows: {
            xs: 2,
            sm: 2,
            md: 2,
            lg: 2,
            xl: 2,
        },
        itemsPerPage: {
            initial: {
                mobile: 8,
                tablet: 12,
                desktop: 24,
            },
            subsequent: {
                mobile: 10,
                tablet: 16,
                desktop: 28,
            },
        },
        relaxedQuery: "",
        includedComponents: [],
        searchCategory: {
            id: "",
            name: "",
        },
        bloomreachCookiesInitialVal: {
            brUid2: "",
            brMtSearch: "",
        },
    },
    viewAllModal: {
        displayName: "",
        name: "",
        isOpen: false,
        activeCharacter: "All",
        facets: {
            opt1: {n: "opt1", c: 1, v: "opt1", incompatibleWith: [], d: false},
            opt2: {n: "opt2", c: 1, v: "opt2", incompatibleWith: ["opt3"], d: false},
            opt3: {n: "opt3", c: 1, v: "opt3", incompatibleWith: ["opt2"], d: false},
        },
        hideSearchFilterModalElements: {
            letterNav: false,
            searchBox: false,
        },
    },
    seoFilters: {
        filterDepthCount: 6,
    },
    features: {
        enablePageInFilters: false,
        enableSearchBanners: false,
        enableTooltips: false,
    },
    categoryQuickLinks: {
        items: [
            {
                href: "/women",
                title: "Women",
                description: "Womens description...",
                imageSrc: "/some/image/src.jpg",
                imageAlt: "women",
            },
            {
                href: "/men",
                title: "Men",
                description: "Mens description...",
                imageSrc: "/some/image/src.jpg",
                imageAlt: "men",
            },
        ],
    },
    tabbedFilters: {
        isOpen: false,
        totalResults: 1234,
        filters: {
            Test1: {
                name: "Test1",
                facets: ["opt1"],
                displayName: "Test 1",
                type: "feat",
                isFilterOpen: true,
                isViewMoreOpen: true,
            },
            Test2: {
                name: "Test2 with options",
                facets: ["opt2"],
                displayName: "Test 2",
                type: "filter",
                isFilterOpen: false,
                isViewMoreOpen: false,
            },
            Test3: {
                name: "Test2 with options",
                displayName: "",
                type: "price",
                isFilterOpen: false,
                max: 100,
                min: 0,
                selectedMax: 70,
                selectedMin: 30,
                currencyCode: "GBP",
            },
            brand: {
                name: "Brand",
                facets: [
                    "brand:znth",
                    "brand:one",
                    "brand:oops",
                    "brand:two",
                    "brand:tcas",
                    "brand:three",
                    "brand:four",
                    "brand:five",
                    "brand:six",
                    "brand:2 size",
                ],
                displayName: "Test 1",
                type: "filter",
                isFilterOpen: true,
                isViewMoreOpen: true,
            },
            Price: {
                name: "Test2 with options",
                displayName: "",
                type: "price",
                isFilterOpen: false,
                max: 100,
                min: 0,
                selectedMax: 70,
                selectedMin: 30,
                currencyCode: "GBP",
            },
        },
        filtersSort: ["Test1", "Test2"],
        facets: {
            opt1: {n: "opt1", c: 1, v: "opt1", incompatibleWith: [], d: false},
            opt2: {n: "opt2", c: 1, v: "opt2", incompatibleWith: ["opt3"], d: false},
            opt3: {n: "opt3", c: 1, v: "opt3", incompatibleWith: ["opt2"], d: false},
        },
        items: [
            {itemNumber: "111111", newIn: true},
            {itemNumber: "222222", newIn: true},
        ],
        selectedFilter: null,
        brandSearch: "",
        historicFacetFilter: {},
        isPerformingKeyFilterSearch: false,
        canSync: true,
        isPriceFacetChanged: false,
    },
}

export const mockTextAlignment = {Value: "ltr"}

export const mockTheme: Theme = {colours: mockColours, dimensions: mockDimensions, direction: "ltr"}
export const mockConfigureStore = configureMockStore([thunk])
export const createMockStoreForRequest = (request: Partial<RequestDuckState>) => {
    return mockConfigureStore({...mockState, request: {...mockState.request, ...request}})
}
const mockStore = mockConfigureStore(mockState)

export default mockStore
