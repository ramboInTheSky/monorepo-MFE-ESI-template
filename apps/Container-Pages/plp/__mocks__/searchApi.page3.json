{
  "totalResults": 96,
  "filters": [
    {
      "name": "Feat",
      "displayName": "",
      "type": "feat",
      "options": [
        {
          "n": "New In",
          "c": 364,
          "v": "newin",
          "s": true
        },
        {
          "n": "In Stock",
          "c": 2405,
          "v": "instock"
        }
      ]
    },
    {
      "name": "Size",
      "displayName": "Size",
      "type": "filter",
      "options": [
        {
          "n": "1 Tall",
          "c": 24,
          "v": "1tall",
          "s": true
        },
        {
          "n": "2",
          "c": 3,
          "v": "2",
          "s": true
        },
        {
          "n": "3",
          "c": 3,
          "v": "3",
          "s": true
        },
        {
          "n": "4 Tall",
          "c": 24,
          "v": "4tall",
          "s": true
        },
        {
          "n": "5",
          "c": 3,
          "v": "5",
          "s": true
        },
        {
          "n": "6 Tall",
          "c": 24,
          "v": "6tall",
          "s": true
        },
        {
          "n": "7",
          "c": 3,
          "v": "7",
          "s": true
        },
        {
          "n": "8 Tall",
          "c": 24,
          "v": "8tall",
          "s": true
        },
        {
          "n": "9",
          "c": 3,
          "v": "9",
          "s": true
        }
      ]
    },
    {
      "name": "Colour",
      "displayName": "Display Colour",
      "type": "filter",
      "options": [
        {
          "n": "8 Tall",
          "c": 24,
          "v": "8tall",
          "s": true
        },
        {
          "n": "18",
          "c": 3,
          "v": "18",
          "s": true
        }
      ]
    },
    {
      "name": "Style",
      "displayName": "Style",
      "type": "filter",
      "options": [
        {
          "n": "Aaaaa Aaaaaa",
          "c": 24,
          "v": "1tall",
          "s": true
        },
        {
          "n": "Bbbbbbbbbb",
          "c": 3,
          "v": "2",
          "s": true
        },
        {
          "n": "Dddddd Dddddd DDddddddddddd",
          "c": 3,
          "v": "3"
        },
        {
          "n": "CCCCC",
          "c": 24,
          "v": "4tall",
          "s": true
        },
        {
          "n": "JJJJJJ",
          "c": 3,
          "v": "5"
        },
        {
          "n": "VVVVV",
          "c": 24,
          "v": "6tall",
          "s": true
        },
        {
          "n": "AAAA",
          "c": 3,
          "v": "7"
        },
        {
          "n": "XXXXX",
          "c": 24,
          "v": "8tall",
          "s": true
        },
        {
          "n": "9",
          "c": 3,
          "v": "9",
          "s": true
        },
        {
          "n": "10",
          "c": 3,
          "v": "10",
          "s": true
        },
        {
          "n": "11",
          "c": 3,
          "v": "11",
          "s": true
        },
        {
          "n": "12",
          "c": 3,
          "v": "12",
          "s": true
        },
        {
          "n": "13",
          "c": 3,
          "v": "13",
          "s": true
        },
        {
          "n": "14",
          "c": 3,
          "v": "14",
          "s": true
        },
        {
          "n": "15",
          "c": 3,
          "v": "15",
          "s": true
        },
        {
          "n": "16",
          "c": 3,
          "v": "16",
          "s": true
        },
        {
          "n": "17",
          "c": 3,
          "v": "17",
          "s": true
        },
        {
          "n": "18",
          "c": 3,
          "v": "18",
          "s": true
        },
        {
          "n": "19",
          "c": 3,
          "v": "19",
          "s": true
        },
        {
          "n": "20",
          "c": 3,
          "v": "20",
          "s": true
        },
        {
          "n": "21",
          "c": 3,
          "v": "21",
          "s": true
        }
      ]
    },
    {
      "name": "Price",
      "displayName": "Price",
      "type": "price",
      "currencyCode": "GBP",
      "min": 0,
      "max": 100,
      "selectedMin": 20,
      "selectedMax": 100
    }
  ],
  "items": [
    {
      "itemNumber": "X3999999",
      "newIn": true
    },
    {
      "itemNumber": "X3800845",
      "newIn": false
    },
    {
      "itemNumber": "X3111111",
      "newIn": false
    },
    {
      "itemNumber": "X32222",
      "newIn": false
    },
    {
      "itemNumber": "X3333",
      "newIn": true
    },
    {
      "itemNumber": "X34444",
      "newIn": false
    },
    {
      "itemNumber": "X35555",
      "newIn": false
    },
    {
      "itemNumber": "X36666",
      "newIn": false
    },
    {
      "itemNumber": "X380077784",
      "newIn": false
    },
    {
      "itemNumber": "X31008884",
      "newIn": true
    },
    {
      "itemNumber": "X32008884",
      "newIn": true
    },
    {
      "itemNumber": "X33008884",
      "newIn": true
    },
    {
      "itemNumber": "X34008884",
      "newIn": true
    },
    {
      "itemNumber": "X35008884",
      "newIn": true
    },
    {
      "itemNumber": "X36008884",
      "newIn": true
    },
    {
      "itemNumber": "X37008884",
      "newIn": true
    },
    {
      "itemNumber": "X37108884",
      "newIn": true
    },
    {
      "itemNumber": "X37208884",
      "newIn": true
    },
    {
      "itemNumber": "X37308884",
      "newIn": true
    },
    {
      "itemNumber": "X37408884",
      "newIn": true
    },
    {
      "itemNumber": "X37508884",
      "newIn": true
    },
    {
      "itemNumber": "X37608884",
      "newIn": true
    },
    {
      "itemNumber": "X37808884",
      "newIn": true
    },
    {
      "itemNumber": "X37908884",
      "newIn": true
    }
  ],
  "pagination": {
    "totalResults": 2403,
    "numberOfItems": 24,
    "start": 0,
    "previousPageUrl": "http://resourceUrl/resourceEndpoint?params",
    "nextPageUrl": "http://resourceUrl/resourceEndpoint?params"
  },
  "keywordRedirect": {
    "redirectedUrl": "/clearance",
    "redirectedQuery": "clearance",
    "originalQuery": "clearance"
  },
  "sorting": {
    "selected": "newpop",
    "options": [
      {
        "name": "Most Relevant",
        "value": "score"
      },
      {
        "name": "Most Popular",
        "value": "newpop"
      },
      {
        "name": "Alphabetical",
        "value": "alpha"
      },
      {
        "name": "Price: Low - High",
        "value": "pricelow"
      },
      {
        "name": "Price: High - Low",
        "value": "pricehi"
      }
    ]
  },
  "redirect": null
}
