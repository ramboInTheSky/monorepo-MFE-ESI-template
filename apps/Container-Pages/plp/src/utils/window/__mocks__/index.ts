import {
    getWindow,
    getDocumentScrollTop,
    removeDocumentScrollListener,
    addDocumentScrollListener,
    isSomewhatInViewport,
    isAboveViewport,
    isBelowViewport,
    IS_BROWSER,
} from ".."

jest.mock("..", () => ({
    getWindow: jest.fn(() => ({})),
    IS_BROWSER: jest.fn(() => true),
    isAboveViewport: jest.fn(),
    isBelowViewport: jest.fn(),
    isSomewhatInViewport: jest.fn(),
    getDocumentScrollTop: jest.fn(),
    addDocumentScrollListener: jest.fn(),
    removeDocumentScrollListener: jest.fn(),
}))

function mockListener(fn: jest.Mock) {
    const api = {
        listener: () => null,
    }
    fn.mockImplementation(listener => {
        api.listener = listener
    })
    return api
}

export function mockGetWindow(value: any = window) {
    ;(getWindow as jest.Mock).mockReturnValue(value)
}

export function mockGetDocumentScrollTop(value: number) {
    ;(getDocumentScrollTop as jest.Mock).mockReturnValue(value)
}

export function mockRemoveDocumentScrollListener() {
    return mockListener(removeDocumentScrollListener as jest.Mock)
}

export function mockAddDocumentScrollListener() {
    return mockListener(addDocumentScrollListener as jest.Mock)
}

export function mockIsSomewhatInViewport(result: boolean) {
    ;(isSomewhatInViewport as jest.Mock).mockReturnValue(result)
}

export function mockIsAboveViewport(result: boolean) {
    ;(isAboveViewport as jest.Mock).mockReturnValue(result)
}

export function mockIsBelowViewport(result: boolean) {
    ;(isBelowViewport as jest.Mock).mockReturnValue(result)
}

export function mockIsBrowser(result: boolean) {
    ;(IS_BROWSER as jest.Mock).mockReturnValue(result)
}
