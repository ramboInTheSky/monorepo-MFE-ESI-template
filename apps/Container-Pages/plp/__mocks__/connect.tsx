import React from "react"
import {Provider} from "react-redux"
import {render} from "@testing-library/react"
import {mockState, mockConfigureStore} from "./mockStore"

interface AnyObject {
    [key: string]: any
}

interface MockComponentProps {
    propsRef: {current: any}
    props: AnyObject
}

const MockComponent = jest.fn(({propsRef, ...props}: MockComponentProps) => {
    // eslint-disable-next-line no-param-reassign
    propsRef.current = props
    return null
}) as any

export function renderConnect(connect: Function, state: AnyObject = {}, otherProps: Record<any, any> = {}) {
    const props = React.createRef()
    const store = mockConfigureStore({...mockState, ...state})
    const ConnectedMock = connect(MockComponent)
    render(
        <Provider store={store}>
            <ConnectedMock {...otherProps} propsRef={props} />
        </Provider>,
    )
    return props.current
}

export function createMockStateForTabbedFilters(searchState: Record<any, any> = {}) {
    return {
        ...mockState,
        tabbedFilters: {
            ...mockState.tabbedFilters,
            ...searchState,
        },
    }
}
