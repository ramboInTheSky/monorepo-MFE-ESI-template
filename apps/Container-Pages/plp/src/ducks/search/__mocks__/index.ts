import {mockState} from "../../../../__mocks__/mockStore"

import {selectHasNextPage, selectHasPreviousPage} from ".."

import {selectFetchTriggerOffset, selectSubsequentItemsPerPage} from "../utils"

jest.mock("../utils", () => ({
    selectFetchTriggerOffset: jest.fn(),
    selectSubsequentItemsPerPage: jest.fn(),
}))
jest.mock("..", () => ({
    fixFacets: jest.fn(() => ({type: "FIX_FACETS"})),
    unfixFacets: jest.fn(() => ({type: "UNFIX_FACETS"})),
    selectItems: jest.fn(),
    selectHasNextPage: jest.fn(),
    selectHasPreviousPage: jest.fn(),
    selectCurrentEndPage: jest.fn(),
    selectCurrentStartPage: jest.fn(),
    selectTotalItemsPerPage: jest.fn(),
    fetchNextPageItemsThunk: jest.fn(() => ({type: "foo"})),
    fetchPreviousPageItemsThunk: jest.fn(() => ({type: "foo"})),
}))

export function mockSelectHasNextPage(result) {
    ;(selectHasNextPage as jest.Mock).mockReturnValue(result)
}

export function mockSelectHasPreviousPage(result) {
    ;(selectHasPreviousPage as jest.Mock).mockReturnValue(result)
}

export function mockSelectFetchTriggerOffset(result) {
    ;(selectFetchTriggerOffset as jest.Mock).mockReturnValue(result)
}
export function mockSelectSubsequentItemsPerPage(result) {
    ;(selectSubsequentItemsPerPage as jest.Mock).mockReturnValue(result)
}

export function createMockStateForSearch(searchState: Record<any, any> = {}) {
    return {
        ...mockState,
        search: {
            ...mockState.search,
            ...searchState,
        },
    }
}
