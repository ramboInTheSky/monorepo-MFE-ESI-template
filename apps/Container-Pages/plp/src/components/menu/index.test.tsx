import React from "react"
import {render, cleanup, fireEvent, screen} from "@testing-library/react"
import {ThemeProvider} from "styled-components"
import {mockText, mockTheme} from "../../../__mocks__/mockStore"
import {Menu} from "."
import TrackSortOpen from "../../events/trackEvent/events/trackSortOpen"

jest.mock("../../events/trackEvent/events/trackSortOpen")

const mockOnSelect = jest.fn()
const componentToTest = (
    <ThemeProvider theme={mockTheme}>
        <Menu
            buttonText="SORT"
            onSelect={mockOnSelect}
            options={{
                selected: "red",
                options: [
                    {name: "red", value: "red"},
                    {name: "blue", value: "blue"},
                ],
            }}
            text={mockText}
        />
    </ThemeProvider>
)
describe("Given a Menu: ", () => {
    afterEach(() => {
        cleanup()
    })

    it("should match the snapshot ", () => {
        const {asFragment} = render(componentToTest)
        expect(asFragment()).toMatchSnapshot()

        expect(screen.queryByTestId("plp-menu-options-sort")).not.toBeInTheDocument()
    })

    it("When clicking the button, it should render the options", () => {
        const {asFragment} = render(componentToTest)
        // test clicking sort opens the expected modal
        fireEvent.click(screen.getByText(/SORT/i))
        expect(asFragment()).toMatchSnapshot()
        expect(screen.getByTestId("plp-menu-options-sort")).toMatchSnapshot()
        expect(screen.queryByText(/SORT/i)).not.toBeInTheDocument()
        expect(TrackSortOpen).toHaveBeenCalled()

        // test tab closes modal
        fireEvent.keyDown(screen.getByTestId("menu-list-grow"), {key: "Tab", code: "KeyTab"})
        expect(screen.getByText(/SORT/i)).toBeInTheDocument()

        // test click outside modal closes modal
        fireEvent.click(screen.getByText(/SORT/i))
        fireEvent.click(screen.getByText(/CLOSE/i))
        expect(screen.getByText(/SORT/i)).toBeInTheDocument()

        // test click an option changes select
        fireEvent.click(screen.getByText(/SORT/i))
        fireEvent.click(screen.getByText(/blue/i))
        expect(screen.getByText(/SORT/i)).toBeInTheDocument()
        expect(mockOnSelect).toHaveBeenCalledWith("blue")
    })
})
