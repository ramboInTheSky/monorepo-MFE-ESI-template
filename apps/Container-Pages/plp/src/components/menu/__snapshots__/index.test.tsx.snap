// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Menu:  When clicking the button, it should render the options 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 {
  width: 100%;
  height: 2.75rem;
  padding: 0;
  text-transform: uppercase;
  background: none;
  background-color: transparent;
  color: #000000;
}

.c1:hover {
  background-color: transparent;
}

.c1.c1:focus {
  border: none;
}

.c2 {
  z-index: 1;
}

.c2[x-placement*="bottom"] .arrow {
  width: 0;
  height: 0;
  border-bottom: 0.625rem solid #000000;
  border-left: 0.625rem solid transparent;
  border-right: 0.625rem solid transparent;
  margin-top: -0.625rem;
}

.c3 {
  position: absolute;
  font-size: 0.4375rem;
  width: 3rem;
  height: 3rem;
}

.c4 {
  border-top: 0.25rem solid #000000;
  margin-top: 0.875rem;
  padding: 0;
}

.c5 {
  padding: 0.8125rem 0.9375rem 0.8125rem 0.625rem;
}

.c6 {
  min-width: 1.875rem;
}

<div
    class="c0"
  >
    <button
      aria-controls="menu-list-grow"
      aria-describedby="styled-options-menu"
      aria-haspopup="true"
      class="MuiButtonBase-root MuiButton-root MuiButton-text c1 MuiButton-disableElevation"
      data-testid="plp-menu-button-sort"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Close
      </span>
    </button>
    <div
      class="c2"
      data-testid="plp-menu-options-sort"
      id="styled-options-menu"
      style="position: absolute;"
      x-placement="bottom"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation3 MuiPaper-rounded"
        style="opacity: 1; transform: scale(1, 1); transform-origin: center top; transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <span
          class="c3 arrow"
        />
        <ul
          class="MuiList-root c4 MuiList-padding"
          data-testid="menu-list-grow"
          id="menu-list-grow"
          role="menu"
          tabindex="-1"
        >
          <li
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root c5 selected Mui-selected MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button Mui-selected Mui-focusVisible Mui-focusVisible"
            role="menuitem"
            tabindex="-1"
          >
            <div
              class="MuiListItemIcon-root c6"
            >
              <img
                alt="check icon"
                src="/static-content/icons/shared/sort-check.svg"
              />
            </div>
            <div
              class="MuiListItemText-root WithStyles(ForwardRef(ListItemText))-root-1"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-h5 MuiTypography-displayBlock"
              >
                red
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </li>
          <li
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root c5 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            role="menuitem"
            tabindex="-1"
          >
            <div
              class="MuiListItemText-root WithStyles(ForwardRef(ListItemText))-root-1 MuiListItemText-inset WithStyles(ForwardRef(ListItemText))-inset-2"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-h5 MuiTypography-displayBlock"
              >
                blue
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Given a Menu:  When clicking the button, it should render the options 2`] = `
.c0 {
  z-index: 1;
}

.c0[x-placement*="bottom"] .arrow {
  width: 0;
  height: 0;
  border-bottom: 0.625rem solid #000000;
  border-left: 0.625rem solid transparent;
  border-right: 0.625rem solid transparent;
  margin-top: -0.625rem;
}

.c1 {
  position: absolute;
  font-size: 0.4375rem;
  width: 3rem;
  height: 3rem;
}

.c2 {
  border-top: 0.25rem solid #000000;
  margin-top: 0.875rem;
  padding: 0;
}

.c3 {
  padding: 0.8125rem 0.9375rem 0.8125rem 0.625rem;
}

.c4 {
  min-width: 1.875rem;
}

<div
  class="c0"
  data-testid="plp-menu-options-sort"
  id="styled-options-menu"
  style="position: absolute;"
  x-placement="bottom"
>
  <div
    class="MuiPaper-root MuiPaper-elevation3 MuiPaper-rounded"
    style="opacity: 1; transform: scale(1, 1); transform-origin: center top; transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <span
      class="c1 arrow"
    />
    <ul
      class="MuiList-root c2 MuiList-padding"
      data-testid="menu-list-grow"
      id="menu-list-grow"
      role="menu"
      tabindex="-1"
    >
      <li
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root c3 selected Mui-selected MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button Mui-selected Mui-focusVisible Mui-focusVisible"
        role="menuitem"
        tabindex="-1"
      >
        <div
          class="MuiListItemIcon-root c4"
        >
          <img
            alt="check icon"
            src="/static-content/icons/shared/sort-check.svg"
          />
        </div>
        <div
          class="MuiListItemText-root WithStyles(ForwardRef(ListItemText))-root-1"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-h5 MuiTypography-displayBlock"
          >
            red
          </span>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </li>
      <li
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root c3 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
        role="menuitem"
        tabindex="-1"
      >
        <div
          class="MuiListItemText-root WithStyles(ForwardRef(ListItemText))-root-1 MuiListItemText-inset WithStyles(ForwardRef(ListItemText))-inset-2"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-h5 MuiTypography-displayBlock"
          >
            blue
          </span>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </li>
    </ul>
  </div>
</div>
`;

exports[`Given a Menu:  should match the snapshot  1`] = `
<DocumentFragment>
  .c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 {
  width: 100%;
  height: 2.75rem;
  padding: 0;
  text-transform: uppercase;
  background: none;
  background-color: transparent;
  color: #000000;
}

.c1:hover {
  background-color: transparent;
}

.c1.c1:focus {
  border: none;
}

<div
    class="c0"
  >
    <button
      aria-haspopup="true"
      class="MuiButtonBase-root MuiButton-root MuiButton-text c1 MuiButton-disableElevation"
      data-testid="plp-menu-button-sort"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        SORT
      </span>
    </button>
  </div>
</DocumentFragment>
`;
